word_language_model/generate.py

```diff
@@ -51,14 +51,22 @@
 ntokens = len(corpus.dictionary)
 hidden = model.init_hidden(1)
 input = torch.randint(ntokens, (1, 1), dtype=torch.long).to(device)
+has_mask = False
 
 with open(args.outf, 'w') as outf:
     with torch.no_grad():  # no tracking history
         for i in range(args.words):
-            output, hidden = model(input, hidden)
-            word_weights = output.squeeze().div(args.temperature).exp().cpu()
-            word_idx = torch.multinomial(word_weights, 1)[0]
-            input.fill_(word_idx)
+            output, hidden = model(input, hidden, has_mask)
+            if model.model_type == 'Transformer':
+                word_weights = output[-1].squeeze().div(args.temperature).exp().cpu()
```

Comment:
Now that we use an nn.Linear decoder I'm surprised you still need to retrieve the last entry.